rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/RestControllerHelper.java

```diff
@@ -58,13 +68,86 @@
     @NonNull
     private final Sw360LicenseService licenseService;
 
+    @NonNull
+    private final ResourceComparatorGenerator resourceComparatorGenerator = new ResourceComparatorGenerator();
+
     private static final Logger LOGGER = Logger.getLogger(RestControllerHelper.class);
 
     public User getSw360UserFromAuthentication(OAuth2Authentication oAuth2Authentication) {
         String userId = oAuth2Authentication.getName();
         return userService.getUserByEmail(userId);
     }
 
+    public PagedResources generatePagesResource(PaginationResult paginationResult, List resources) throws URISyntaxException {
+        PaginationOptions paginationOptions = paginationResult.getPaginationOptions();
+        List<Link> pagingLinks = this.getPaginationLinks(paginationResult, this.getAPIBaseUrl());
+        PagedResources.PageMetadata pageMetadata = new PagedResources.PageMetadata(
+                paginationOptions.getPageSize(),
+                paginationOptions.getPageNumber(),
+                paginationResult.getTotalCount(),
+                paginationResult.getTotalPageCount());
+        return new PagedResources<>(resources, pageMetadata, pagingLinks);
+    }
+
+    private List<Link> getPaginationLinks(PaginationResult paginationResult, String baseUrl) {
+        PaginationOptions paginationOptions = paginationResult.getPaginationOptions();
+        List<Link> paginationLinks = new ArrayList<>();
+
+        paginationLinks.add(new Link(createPaginationLink(baseUrl, 0, paginationOptions.getPageSize()),"first"));
```

Comment:
Maybe extract the String previos, next, last... ?page and &page_entries to const.